source/common/router/router.h

```diff
@@ -384,6 +384,7 @@ class Filter : Logger::Loggable<Logger::Id::router>,
   const Network::Connection* downstreamConnection() const override {
     return callbacks_->connection().ptr();
   }
+  StreamInfo::StreamInfo* downstreamInfo() const override { return &callbacks_->streamInfo(); }
```

Comment:
This definitely makes me twitchy.
For L7, we have a patterm of when we need to of get upstream information to the downstream stream, we do by putting it in the upstream stream info and copying or moving to the downstream stream info when we associate the connections. I'm concerned if we add this to the API we're going to end up with lifetime issues.
Can instead pass the downstream info as a constructor to HttpConnPool so the addition is limited to TCP proxying?

Reply:
Yes, I wasn't happy about it either. Let me explore a better way. FWIW it's already available as immutable via `downstreamConnection` (so the issue is already present if downstream dies before upstream).